Kernel/Signature.hpp

```diff
@@ -298,20 +300,43 @@ class Signature
     inline bool realConstant() const
     { return interpreted() && arity()==0 && fnType()->result()==AtomicSort::realSort(); }
 
+  private:
+    bool numeralConstant(RealConstantType*) const { return realConstant(); }
+    bool numeralConstant(RationalConstantType*) const { return rationalConstant(); }
+    bool numeralConstant(IntegerConstantType*) const { return integerConstant(); }
+  public:
+
+    /** template version of {integer,rational,real}Constant */
+    template<class Number> bool numeralConstant() const
+    { return numeralConstant((Number*)nullptr); }
+
     /** return true if an interpreted number, note subtle but significant difference from numericConstant **/
     inline bool interpretedNumber() const
```

Comment:
Sorry, I think I asked before. But there is no `significant difference from numericConstant` anymore, right? (after all, there does not seem to be `numericConstant` anymore).

Reply:
indeed. I updated the outdated comment.